### PR TITLE
Add GSoC 2022 landing page and announcement

### PR DESCRIPTION
**Pull request by lmmentel**
##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/alan-turing-institute/sktime/blob/main/.all-contributorsrc).
- [ ] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/alan-turing-institute/sktime/blob/main/CODEOWNERS) to receive notifications about future changes to these files.
- [ ] I've added unit tests and made sure they pass locally.
- [ ] The PR title starts with either [ENH], [DOC] or [BUG] indicating wether the PR topic is related to enhancement, documentation or bug

##### For new estimators
- [ ] I've added the estimator to the online documentation.
- [ ] I've updated the existing example notebooks or provided a new one to showcase how my estimator works.

### CODE REVIEW

**Comment by fkiraly** [comment]
we should not duplicate the page, is there a way to have a single source of truth?

**Review by fkiraly — CHANGES_REQUESTED** [review]
some comments, for discussion:
* I wouldn't put it at the top of the *docs*, if at all at the top of README (we want to target developers, not users), or the "mentoring" page
* seems to duplicate content in the mentoring repository. Can we avoid forking/duplication?

**Comment by fkiraly** [comment]
I like this, so I´d tend to merge if there´s no obvious/easy way to resolve the duplication - any good ideas?

File: docs/source/index.rst
Comment (Lovkush-A) [inline docs/source/index.rst:20]: is it possible to link to the markdown file: https://github.com/sktime/mentoring/blob/main/internships/gsoc2022.md

Otherwise it requires having to update two files for any change.

Comment (lmmentel) [reply docs/source/index.rst:20]: Probably possible but not worth the hassle, I can update the docs manually if you ping me :)

**Comment by lmmentel** [comment]
@fkiraly the main page  is just temporary change until we close the application process, but with the speed at which we move it might not be relevant anymore by the time we merge.

Since this is not code but information  duplication is fine as far as I'm concerned. More ways to get to what people might be searching for.

Not sure if I put any more time into this given the date.

**Comment by fkiraly** [comment]
@lmmentel, this will only go live at the next version, not at PR, so anything until Sat will be fine.

Could I suggest to proceed as follows: kindly get in touch with @Lovkush-A who currently handles the original document which this duplicates. If he confirms that your version is ok, and is otherwise happy, he should approve. In that case, I will also approve on the basis that the process owner has approved.

**Comment by lmmentel** [comment]
@Lovkush-A any further thoughts?

**Comment by Lovkush-A** [comment]
I am happy with Franz's plan. Any new details that emerge will be about interview stage, and we will be emailing the candidates.

However, before merging, this pr needs to be updated with the latest version (e.g. this pr does not have the instructinos on getting feedback).

**Comment by lmmentel** [comment]
@Lovkush-A I updated the docs :slightly_smiling_face:

**Review by Lovkush-A — APPROVED** [review]
happy to approve

**Review by fkiraly — APPROVED** [review]
As discussed above, since @Lovkush-A approves, I also approve.

**Comment by fkiraly** [comment]
readthedocs is failing - I think for this PR, we want it to pass?

**Comment by lmmentel** [comment]
> readthedocs is failing - I think for this PR, we want it to pass?

I see a green tick mark below.

**Pull request merged**
